### PR TITLE
Prom linear elasticity

### PR DESCRIPTION
**Pull request by bumpwy**
### 1. first two offline phases for `nu=0.2 & 0.4`
`python linear_elasticity_global_rom.py -offline -id 0 -nu 0.2`
`python linear_elasticity_global_rom.py -offline -id 1 -nu 0.4`
### 2. the merge phase:
`python linear_elasticity_global_rom.py -merge -ns 2`
### 3. final offline phase FOM for `nu=0.3`
`python linear_elasticity_global_rom.py -offline -id 2 -nu 0.3`
results:
> PCG Iterations = 17
> Final PCG Relative Residual Norm = 4.35538e-09
> Creating file: basis2_snapshot.000000
> Elapsed time for assembling FOM: 0.002839803695678711 second
> Elapsed time for solving FOM: 0.005137205123901367 second

### 4. online phase ROM for `nu=0.3`
`python linear_elasticity_global_rom.py -online -id 3 -nu 0.3`
results:
> Size of linear system: 1170
> Opening file: basis.000000
> On rank 0, spatial basis dimension is 1170 x 2
> Relative error of ROM for E=2.5 and nu=0.3 is 0.0005710430923335614
> Elapsed time for assembling ROM: 0.02164936065673828 second
> Elapsed time for solving ROM: 1.1920928955078125e-05 second

### CODE REVIEW

**Review by dreamer2368 — APPROVED** [review]
Running on quartz on my end showed the consistent result as shown in PR.

File: examples/prom/linear_elasticity_global_rom.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+# 8. Determine the list of true (i.e. parallel conforming) essential
+#    boundary dofs. In this example, the boundary conditions are defined by
+#    marking only boundary attribute 1 from the mesh as essential and
+#    converting it to a list of true dofs.
+ess_bdr = mfem.intArray(pmesh.bdr_attributes.Max())
+ess_tdof_list = mfem.intArray()
+ess_bdr.Assign(0)
+ess_bdr[0] = 1
+fespace.GetEssentialTrueDofs(ess_bdr, ess_tdof_list)
+print("here")
```
Comment (siuwuncheung) [inline examples/prom/linear_elasticity_global_rom.py]: I guess it can be removed?

File: examples/prom/linear_elasticity_global_rom.py [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+    f.Set(i, mfem.ConstantCoefficient(0.0))
+
+pull_force = mfem.Vector([0]*pmesh.bdr_attributes.Max())
+pull_force[1] = ext_force
+f.Set(dim-1, mfem.PWConstCoefficient(pull_force))
+
+
+b = mfem.ParLinearForm(fespace)
+b.AddBoundaryIntegrator(mfem.VectorBoundaryLFIntegrator(f))
+print('r.h.s. ...')
```
Comment (siuwuncheung) [inline examples/prom/linear_elasticity_global_rom.py]: Consider removing or adding `if myid == 0:`?

**Review by siuwuncheung — COMMENTED** [review]
Thanks @bumpwy! Look good to me overall.

**Comment by bumpwy** [comment]
Hi @siuwuncheung thanks for the review I have edited the code accordingly.

**Review by siuwuncheung — APPROVED** [review]

**Pull request merged**